Bugzilla/WebService/User.pm

```diff
@@ -504,6 +505,63 @@ sub whoami {
   );
 }
 
+#
+# Saved Searches
+#
+
+sub add_saved_search {
+  my ($self, $params) = @_;
+
+  # The `create()` method fails when a saved search with the same name exists.
+  # The UI should ask in advance if the user wants to override the existing one,
+  # and if the answer is yes, use `update_saved_search()` instead.
+  my $search = Bugzilla::Search::Saved->create({
+    name => $params->{name}, query => $params->{url}, link_in_footer => 1
```

Comment:
Here is a bit of danger. It might be possible for the `$params->{url}` to be a data structure.
It's not certain this would cause a problem. `Bugzilla::Search::Saved->create($user_input)` would definitely be bad. Still, it's always good to be explicit about what something is.

My other review comments add type checking but I didn't proof them so they may need to be tweaked.